tools/fs/FS_Dump_Parser/src/smartfs_dump_parser/parts/DumpViewer.java

```diff
@@ -0,0 +1,276 @@
+		return tableViewer;
+	}
+
+	public TreeViewer getDirectoryViewer() {
+		return treeViewer;
+	}
+
+	private void createDirectoryTreeViewer(final Composite parent) {
+		treeViewer = new TreeViewer(parent, SWT.MULTI | SWT.H_SCROLL | SWT.V_SCROLL | SWT.FULL_SELECTION | SWT.BORDER);
+
+		treeViewer.setContentProvider(new ITreeContentProvider() {
+			@Override
+			public Object[] getElements(Object inputElement) {
+				return getChildren(inputElement);
+			}
+
+			@Override
+			public Object[] getChildren(Object parentElement) {
+				return ((SmartFile) parentElement).getEntries().toArray();
+			}
+
+			@Override
+			public Object getParent(Object element) {
+				return ((SmartFile) element).getParent();
+			}
+
+			@Override
+			public boolean hasChildren(Object element) {
+				return !(((SmartFile) element).getEntries().isEmpty());
+			}
+		});
+
+		treeViewer.setLabelProvider(new LabelProvider() {
+			@Override
+			public String getText(Object element) {
+				return ((SmartFile) element).getFileName();
+			}
+		});
+
+		treeViewer.setInput(ModelProvider.getTopDummyDirectory());
+
+		GridData gridData = new GridData();
+		gridData.verticalAlignment = GridData.FILL;
+		gridData.horizontalSpan = 3;
+		gridData.grabExcessHorizontalSpace = true;
+		gridData.horizontalAlignment = GridData.FILL;
+		treeViewer.getControl().setLayoutData(gridData);
+	}
+
+	private void createCheckbox(final Composite parent) {
+		activeButton = new Button(parent, SWT.CHECK);
+		activeButton.setText("Active Sectors");
+		activeButton.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				if (activeButton.getSelection()) {
+					if (ModelProvider.INSTANCE.getSectors().size() == 0) {
+						MessageDialog.openError(parent.getShell(), "Error", "Please open a dump file first.");
+						activeButton.setSelection(false);
+						return;
+					}
+					List<Sector> activeSectors = ModelProvider.INSTANCE.getActiveSectors();
+					if (activeSectors == null) {
+						activeSectors = SmartFSOrganizer.filterSectors(SectorStatus.ACTIVE,
+								ModelProvider.INSTANCE.getSectors());
+						ModelProvider.INSTANCE.setActiveSectors(activeSectors);
+					}
+					tableViewer.setInput(activeSectors);
+					tableViewer.refresh();
+
+					dirtyButton.setSelection(false);
+					cleanButton.setSelection(false);
+				}
+			}
+		});
+
+		dirtyButton = new Button(parent, SWT.CHECK);
+		dirtyButton.setText("Dirty Sectors");
+		dirtyButton.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				if (ModelProvider.INSTANCE.getSectors().size() == 0) {
+					MessageDialog.openError(parent.getShell(), "Error", "Please open a dump file first.");
+					dirtyButton.setSelection(false);
+					return;
+				}
+				List<Sector> dirtySectors = ModelProvider.INSTANCE.getDirtySectors();
+				if (dirtySectors == null) {
+					dirtySectors = SmartFSOrganizer.filterSectors(SectorStatus.DIRTY,
```

Comment:
Will you make list when each of button pressed?
After load data from flash, you can make list immediately. you can input pre-defined 3 type of list,
```
if (isObsoledted(Cursector)) {
     mObsoletedList.add(Cursector);
} else if ....
```

Reply:
Okay, modified.